### ISSUE TITLE
WPF UI Doesn't seem to render release notes

### ISSUE THREAD

**Issue by jfro** [NetSparkleUpdater/NetSparkle#116]
Poked at it some, and I see valid looking HTML going to the UI's WebBrowser control, but it remains blank. Seems like the code is all there and content is being built for it, so figure there must be a bug somewhere. Wasn't able to notice anything myself though.

I also didn't see the loading release notes message show at all before my content is attempted to be loaded.

**Comment by Deadpikle** [comment]
Hi @jfro! Thanks for the report. This is working fine in the samples.

Please provide a reproducible sample if you can. Can you point me to your appcast? Are you using a WPF UI library that might be messing with this? Can you download the code here and try sticking your appcast URL in the samples to see if it is working there?

Thanks!

**Comment by jfro** [comment]
Hmm samples timed out on the default setup, will try again. Feeding the WPF sample app my appcast URL reproduced it so that's promising. I'll investigate further and get you more info soon

**Comment by jfro** [comment]
Here's a copy of the appcast, it's a non-public build currently so just stripped the base URL, otherwise the appcast is what's being served to the updater. It pulls it down OK and generates HTML from what I can see. We used generate_appcast with a directory of markdown files & our installer files
[studio.zip](https://github.com/NetSparkleUpdater/NetSparkle/files/4913016/studio.zip)

**Comment by jfro** [comment]
Narrowed it down to inline changelog contents via <description> tags. This is what generate_appcast did when pointed at folder of markdown.

There a way to use that tool with the URL style changelog like your sample has?

**Comment by jfro** [comment]
OK figured out the URL method, wasn't clear to me from --help.

So I'd say this bug is narrowed down to just non-URL changelog path when using appcast_generator

**Comment by Deadpikle** [comment]
@jfro,

Thanks so much for doing some digging and figuring things out. I really appreciate that.

So, sounds like there are 2 action items here:

1. `--help` needs to be updated to show more info about the URL thingy (and/or more examples -- we used to have these but they were removed with the rewrite).
2. Non-URL changelog items are not being loaded properly in `ReleaseNotesGrabber`.

**Comment by Deadpikle** [comment]
OK, the issue was much bigger than just non-URL changelog items not loading. There's actually some sort of race condition on creating the window and loading something in the PWF `WebBrowser` -- if you tell the WPF WebBrowser to navigate to the release notes string too quickly, it freaks out and doesn't load anything. If you tell it to navigate to `about:blank` first, then navigate to the release notes when that first step is done, it loads things fine. (Got the idea for this after looking at the .NET Framework code: https://referencesource.microsoft.com/#PresentationFramework/src/Framework/System/Windows/Controls/WebBrowser.cs,2ef8dace9bb8d839)

I've made a fix for this that seems to work, but I'm not convinced it's the right idea...it would explain why sometimes the notes wouldn't show up even for URL-based changelogs -- things were just loading too quickly. See 2.0.0-preview20200726001.

Perhaps the real fix is to write release notes to disk in the user's Temp folder and then read from disk??? 🤔

This only affects WPF -- WinForms was working fine.

-----

Re: Docs -- I've updated #89 to remind me/whomever that there needs to be more docs updated in the app cast generator area. I also clarified the difference between `ChangeLogUrl` and `ChangeLogPath` in the actual command line app for now.
